src/yarp.c

```diff
@@ -9805,6 +9805,17 @@ parse_arguments_list(yp_parser_t *parser, yp_arguments_t *arguments, bool accept
     return found;
 }
 
+// Checks whether the predicate for a conditional is with expected tokens.
+static inline void
+check_predicate_closed(yp_parser_t *parser) {
```

Comment:
Could we actually make this `parse_predicate` and include `yp_node_t *predicate = parse_expression(parser, YP_BINDING_POWER_MODIFIER, error_id);` and pushing the context? (It looks like we're missing the `context_push` as well.

Reply:
Done in https://github.com/ruby/prism/pull/1577/commits/bf43006d0a0644fa46f1ca196e26d6dd887afbf8